framework/decode/dx12_replay_consumer_base.cpp

```diff
@@ -1204,6 +1214,17 @@ HRESULT Dx12ReplayConsumerBase::OverrideCreateCommittedResource(
 
     auto clear_value_pointer = pOptimizedClearValue->GetPointer();
 
+    D3D12_RESOURCE_STATES modified_initial_resource_state = InitialResourceState;
+    if ((options_.dump_resources_type != DumpResourcesType::kNone) &&
+        (modified_initial_resource_state & D3D12_RESOURCE_STATE_ALL_SHADER_RESOURCE))
+    {
+        // shader resources can't change state after SetGraphicsRootDescriptorTable.
+        // But even if it changes state and copy resources before SetGraphicsRootDescriptorTable,
+        // it can't solve it, because it also has to copy resources after drawcall.
+        // In this case, adding a COPY_SOURCE state here, so it doesn't need to change state for copy resources.
+        modified_initial_resource_state |= D3D12_RESOURCE_STATE_COPY_SOURCE;
```

Comment:
shader resources can't change state after SetGraphicsRootDescriptorTable. In this case, adding a COPY_SOURCE state for shader resource, so it doesn't need to change state for copy resources.